pipeline-runner/R/gem2s-6-construct_qc_config.R

```diff
@@ -159,7 +159,15 @@ get_sample_mitochondrial_config <- function(scdata_list.sample, config) {
 
 # threshold for doublet score is the max score given to a singlet (above score => doublets)
 get_dblscore_config <- function(scdata_list, config) {
-  probabilityThreshold <- max(scdata_list$doublet_scores[scdata_list$doublet_class == "singlet"], na.rm = TRUE)
+  # if no singlets, set threshold to 0
+  # this is required to prevent the threshold to be "-Inf" when there are no singlets left causing subset experiment to fail
+  is_singlet <- scdata_list$doublet_class == "singlet"
+  if (all(is_singlet == FALSE)) {
+    probabilityThreshold <- 0.0
+  } else {
+    probabilityThreshold <- max(scdata_list$doublet_scores[is_singlet], na.rm = TRUE)
+  }
```

Comment:
you could do this instead, and get rid of the if/else

```suggestion
  is_singlet <- scdata_list$doublet_class == "singlet"
  probabilityThreshold <- max(scdata_list$doublet_scores[is_singlet], 0.0, na.rm = TRUE)
```